packages/server-core/src/bot/bot/bot.hooks.ts

```diff
@@ -39,7 +44,29 @@ import {
   botResolver
 } from '../../bot/bot/bot.resolvers'
 import authenticate from '../../hooks/authenticate'
+import persistData from '../../hooks/persist-data'
 import verifyScope from '../../hooks/verify-scope'
+import { BotService } from './bot.class'
+
+async function sanitizeInstanceId(context: HookContext<BotService>) {
+  if (Array.isArray(context.data) || context.method !== 'create') {
+    throw new BadRequest(`${context.path} service only works for single object create`)
+  }
+
+  context.data!.instanceId = context.data?.instanceId ?? ('' as InstanceID)
+}
+
+async function createAndPopulateBotCommands(context: HookContext<BotService>) {
```

Comment:
This should be named `addBotCommands` to make it similar convention following in other hooks.